assets/app/scripts/controllers/overview.js

```diff
@@ -231,14 +233,43 @@ angular.module('openshiftConsole')
       Logger.log("deployments (subscribe)", $scope.deployments);
     }));
 
-    // Sets up subscription for images and imagesByDockerReference
-    watches.push(DataService.watch("images", $scope, function(images) {
-      $scope.images = images.by("metadata.name");
-      $scope.imagesByDockerReference = images.by("dockerImageReference");
-      Logger.log("images (subscribe)", $scope.images);
-      Logger.log("imagesByDockerReference (subscribe)", $scope.imagesByDockerReference);
+    // Sets up subscription for imageStreams
+    watches.push(DataService.watch("imageStreams", $scope, function(imageStreams) {
+      $scope.imageStreams = imageStreams.by("metadata.name");
+      buildDockerRefMapForImageStreams();
+      fetchReferencedImages();
+      // TODO look for "referenced images"
+      Logger.log("imageStreams (subscribe)", $scope.imageStreams);
     }));
 
+    var fetchReferencedImages = function() {
+      angular.forEach($scope.pods, function(pod){
+        angular.forEach(pod.spec.containers, function(container){
+          var dockerRef = container.image;
+          if (!$scope.imagesByDockerReference[dockerRef] && $scope.imageStreamImageRefByDockerReference[dockerRef]) {
+            DataService.get("imageStreamImages", $scope.imageStreamImageRefByDockerReference[container.image], $scope).then(
+              function(image) {
+                $scope.imagesByDockerReference[dockerRef] = image;
+              }
+            );
+          }
+        });
+      })
+    };
+
+    var buildDockerRefMapForImageStreams = function() {
+      angular.forEach($scope.imageStreams, function(imageStream) {
+        angular.forEach(imageStream.status.tags, function(tag){
+          angular.forEach(tag.items, function(image){
+            if (image.image) {
+              $scope.imageStreamImageRefByDockerReference[image.dockerImageReference] = imageStream.metadata.name + "@" + image.image;
```

Comment:
Yeah, making clients concatenate stuff themselves isn't good... better to give them an opaque value they can directly compare for equality